server/src/com/thoughtworks/go/server/scheduling/BuildCauseProducerService.java

```diff
@@ -246,6 +251,46 @@ ServerHealthState newProduceBuildCause(PipelineConfig pipelineConfig, BuildType
         }
     }
 
+    private boolean isGoodReasonToSchedule(PipelineConfig pipelineConfig, BuildCause buildCause, BuildType buildType,
+                                           boolean materialConfigurationChanged,MaterialRevisions peggedRevisions) {
+        if(buildCause == null)
+            return false;
+
+        boolean validCause = buildType.isValidBuildCause(pipelineConfig, buildCause);
+
```

Comment:
Any specific reason for ignoring `materialConfigurationChanged` flag when `pipelineConfig.isConfigOriginSameAsOneOfMaterials()` == true?
If not, would rewriting this if-else block to be something like this make sense:

```
        if (pipelineConfig.isConfigOriginSameAsOneOfMaterials()) {
            // then we need config and material revisions to be consistent
            if (!buildCause.isForced() && !buildCause.pipelineConfigAndMaterialRevisionMatch(pipelineConfig)) {
                return false;
            }
        } else {
            return materialConfigurationChanged || validCause;
        }
```